cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/main/ProgramRunInfoSerializer.java

```diff
@@ -58,31 +60,47 @@
 
   /**
-   * convert {@link ProgramRunInfo} to a Generic record
+   * Converts {@link ProgramRunInfo} to a Generic record
    */
-  public static GenericData.Record createRecord(ProgramRunInfo runIdFields) {
+  public static GenericData.Record createRecord(ProgramRunInfo runInfo) {
     GenericData.Record startInfoRecord = null;
-    if (runIdFields.getProgramStatus().equals("STARTING")) {
+    if (runInfo.getProgramStatus().equals("STARTING")) {
       startInfoRecord = new GenericData.Record(STARTING_INFO_SCHEMA);
-      startInfoRecord.put(Constants.USER, runIdFields.getProgramSartInfo().getPrincipal());
-      startInfoRecord.put(Constants.RUNTIME_ARGUMENTS, runIdFields.getProgramSartInfo().getRuntimeArguments());
+      startInfoRecord.put(Constants.USER, runInfo.getProgramSartInfo().getPrincipal());
+      startInfoRecord.put(Constants.RUNTIME_ARGUMENTS, runInfo.getProgramSartInfo().getRuntimeArguments());
       GenericData.Record artifactRecord = new GenericData.Record(ARTIFACT_INFO_SCHEMA);
-      ArtifactId artifactId = runIdFields.getProgramSartInfo().getArtifactId();
+      ArtifactId artifactId = runInfo.getProgramSartInfo().getArtifactId();
       artifactRecord.put(Constants.ARTIFACT_NAME, artifactId.getName());
       artifactRecord.put(Constants.ARTIFACT_VERSION, artifactId.getVersion().toString());
       artifactRecord.put(Constants.ARTIFACT_SCOPE, artifactId.getScope().toString());
       startInfoRecord.put(Constants.ARTIFACT_ID, artifactRecord);
     }
     GenericData.Record record = new GenericData.Record(SCHEMA);
-    record.put(Constants.NAMESPACE, runIdFields.getNamespace());
-    record.put(Constants.APPLICATION_NAME, runIdFields.getApplication());
-    record.put(Constants.APPLICATION_VERSION, runIdFields.getApplicationVersion());
-    record.put(Constants.PROGRAM, runIdFields.getProgram());
-    record.put(Constants.RUN, runIdFields.getRun());
-    record.put(Constants.STATUS, runIdFields.getProgramStatus());
-    record.put(Constants.TIME, runIdFields.getTimestamp());
-    record.put(Constants.MESSAGE_ID, runIdFields.getMessageId());
+    record.put(Constants.NAMESPACE, runInfo.getNamespace());
+    record.put(Constants.APPLICATION_NAME, runInfo.getApplication());
+    record.put(Constants.APPLICATION_VERSION, runInfo.getApplicationVersion());
+    record.put(Constants.PROGRAM_TYPE, runInfo.getType());
+    record.put(Constants.PROGRAM, runInfo.getProgram());
+    record.put(Constants.RUN, runInfo.getRun());
+    record.put(Constants.STATUS, runInfo.getProgramStatus());
+    record.put(Constants.TIME, runInfo.getTimestamp());
+    record.put(Constants.MESSAGE_ID, runInfo.getMessageId());
     record.put(Constants.START_INFO, startInfoRecord);
     return record;
   }
+
+  /**
+   * Creates a report record with the given information
+   */
+  @VisibleForTesting
+  public static GenericData.Record createRecord(String namespace, String application, String version,
```

Comment:
if this method is used only in test case, then it should be part of the test case/package. `@VisibleForTesting`annotation is used as convention only for package private methods, as a way to indicate they are called from the test class of the same package. they are not used for public static methods.